extensions/wikia/EmbeddableDiscussions/EmbeddableDiscussionsController.class.php

```diff
@@ -56,35 +57,52 @@ public static function render( $input, array $args, Parser $parser, PPFrame $fra
 		if ( F::app()->checkSkin( 'wikiamobile' ) ) {
 			// In Mercury, discussions are rendered client side as an Ember component
 			$modelData = [
-				'show' => $showLatest ? 'latest' : 'trending',
-				'itemCount' => $itemCount,
+				'mercuryComponentAttrs' => json_encode( [
+					'category' => $category,
+					'show' => $showLatest ? 'latest' : 'trending',
+					'itemCount' => $itemCount,
+				] ),
+				'loading' => wfMessage( 'embeddable-discussions-loading' )->plain()
 			];
 
-			// In mercury, discussions app is rendered client side
+			// In mercury, discussions app is rendered client side in an Ember container
 			$html = $templateEngine->clearData()
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Immediately return this expression instead of assigning it to the temporary variable "$html". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1488)